x/tbr/genesis.go

```diff
@@ -2,48 +2,46 @@ package tbr
 
 import (
 	"errors"
-	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 type GenesisState struct {
-	RewardDenom string       `json:"reward_denom"`
-	PoolAmount  sdk.DecCoins `json:"pool_amount"`
+	PoolAmount       sdk.DecCoins `json:"pool_amount"`
+	YearlyPoolAmount sdk.DecCoins `json:"yearly_pool_amount"`
+	YearNumber       int64        `json:"year_number"`
 }
 
 // DefaultGenesisState returns a default genesis state
-func DefaultGenesisState(rewardDenom string) GenesisState {
-	return GenesisState{
-		RewardDenom: rewardDenom,
-		PoolAmount:  sdk.DecCoins{},
-	}
+func DefaultGenesisState() GenesisState {
+	return GenesisState{}
 }
 
 // InitGenesis sets the initial Block Reward Pool amount for genesis.
 func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) {
 	if data.PoolAmount != nil {
-		keeper.SetBlockRewardsPool(ctx, data.PoolAmount)
+		keeper.SetTotalRewardPool(ctx, data.PoolAmount)
+		keeper.SetYearlyRewardPool(ctx, data.PoolAmount.QuoDec(sdk.NewDec(5)))
+	}
+
+	if data.YearlyPoolAmount != nil {
+		keeper.SetYearlyRewardPool(ctx, data.YearlyPoolAmount)
 	}
 
-	keeper.SetRewardDenom(ctx, data.RewardDenom)
+	keeper.SetYearNumber(ctx, data.YearNumber)
 }
 
 // ExportGenesis returns a GenesisState for a given context and keeper.
 func ExportGenesis(ctx sdk.Context, keeper Keeper) GenesisState {
 	return GenesisState{
-		RewardDenom: keeper.GetRewardDenom(ctx),
-		PoolAmount:  keeper.GetBlockRewardsPool(ctx),
+		PoolAmount: keeper.GetTotalRewardPool(ctx),
+		YearNumber: keeper.GetYearNumber(ctx),
```

Comment:
Here you are not exporting `YearlyPoolAmount` and above in the `InitGenesis` its value will be `nil` everytime, so the `YearlyPoolAmount` will always be set with `data.PoolAmount`. I don't know if this was done by will or not. I'll let this comment here to let you know if maybe you didn't want to do this.

Reply:
Good catch, I didn't see it. Added the export :rocket: